relay-server/src/service.rs

```diff
@@ -85,12 +85,13 @@ pub fn create_runtime(name: &str, threads: usize) -> Runtime {
         // Relay also does not use other blocking opertions from Tokio which require
         // this pool, no usage of `tokio::fs` and `tokio::io::{Stdin, Stdout, Stderr}`.
         //
-        // We limit the maximum amount of threads here, we've seen that Tokio
-        // expands this pool very very aggressively and basically never shrinks it
-        // which leads to a massive resource waste.
-        .max_blocking_threads(150)
```

Comment:
I would still keep the limit, I don't think there is a reasonable explanations to ever need more than this. At least until we verified the semaphore does what we hope it to do.

Reply:
Given the amount of threads running redis blocking code is now dependant on the pool size, I'm not completely sold in picking an arbitrary number X given we still don't know how much 3rd party code might rely on that..
In short: why is 150 better than 512?

Reply:
512 will OOM, 150 does not. Since neither threshold should ever be hit in the first place, I prefer taking the smaller one (could even be lower) which won't make Relay OOM.

Reply:
Understand, by why the memory usage gets so high? Theoretically the increase should only reflects what we do in the closures passed to `spawn_blocking`..

Anyways, my point is that the 150 value is based on our SaaS deployments of Relay, whereas self-hosted instances might have different running conditions. Maybe we should make that value configurable and use 150 as a default?

Reply:
I think we're overthinking this massively. I am all for removing this limit again, if we know for sure the semaphore 'fixes' the problem we're seeing, I'd just want to do it after we know.

150 is more than enough in our SaaS environment (we have the metrics to back this up), self hosted should never even come close to the load we have running in SaaS (they don't have the amount of projects we have, even if they have the same req/s per instance) -> 150 satisfies these needs.